src/sagas/repository.js

```diff
@@ -158,20 +169,52 @@ export function* getDatasetPolicy({ payload }) {
   }
 }
 
-export function* addReadersToDataset({ payload }) {
-  // TODO is this going to need a loop for the list?
-  const datasetId = payload.jobResult.id;
-  const readersList = yield select(getReaders);
+export function* addReadersDataset(datasetId, reader) {
+  const readerObject = { email: reader };
   try {
     const response = yield call(
       authPost,
-      '/api/repository/v1/datasets/' + datasetId + '/policies/readers/members',
-      readersList,
+      '/api/repository/v1/datasets/' + datasetId + '/policies/reader/members',
+      readerObject,
     );
     yield put({
       type: ActionTypes.SET_DATASET_POLICY_SUCCESS,
       dataset: { data: response },
     });
+    yield call(getDatasetPolicy, { payload: datasetId });
+  } catch (err) {
+    yield put({
+      type: ActionTypes.EXCEPTION,
+      payload: err,
+    });
+  }
+}
+
+export function* setDatasetPolicy({ payload }) {
+  const datasetId = payload.datasetId;
+  const reader = payload.users[0];
+  yield call(addReadersDataset, datasetId, reader);
+}
+
+export function* createDatasetPolicy({ payload }) {
+  const datasetId = payload.jobResult.id;
+  const readerList = yield select(getReaders);
+  for (let i = 0; i < readerList.length; i++) {
+    yield put(addReadersDataset, datasetId, readerList[i]);
+  }
+}
+
+export function* removeReaderFromDataset({ payload }) {
+  const datasetId = payload.datasetId;
+  const reader = payload.user;
+  const url = '/api/repository/v1/datasets/' + datasetId + '/policies/reader/members/' + reader;
+  try {
+    const response = yield call(authDelete, url);
+    yield put({
+      type: ActionTypes.REMOVE_READER_FROM_DATASET_SUCCESS,
+      dataset: { data: response },
+    });
+    yield call(getDatasetPolicy, { payload: datasetId });
```

Comment:
we also re-fetch for you on delete member

Reply:
We'll go through each of these during pairing on Tuesday! I think we should make the changes, and I can walk you thru the code then